packages/grid/x-data-grid/src/internals/components/virtualization/GridVirtualScrollerContent.tsx

```diff
@@ -30,13 +30,18 @@ const GridVirtualScrollerContent = React.forwardRef<
   HTMLDivElement,
   React.HTMLAttributes<HTMLDivElement>
 >(function GridVirtualScrollerContent(props, ref) {
-  const { className, ...other } = props;
+  const { className, style, ...other } = props;
   const rootProps = useGridRootProps();
-  const ownerState = { classes: rootProps.classes };
+  const ownerState = { classes: rootProps.classes, overflowedContent: style?.minHeight === 'auto' };
```

Comment:
I noticed that `minHeight` is not reliable when `autoHeight` is on. Removing a row and re-adding it in https://deploy-preview-3519--material-ui-x.netlify.app/components/data-grid/layout/#auto-height, adds the overflow class. I propose to use:

```suggestion
  const ownerState = { classes: rootProps.classes, overflowedContent: !rootProps.autoHeight && style?.minHeight === 'auto' };
```

The content never overflows when `autoHeight` is on.

![image](https://user-images.githubusercontent.com/42154031/155384040-c36982a8-02a5-446e-9215-0d62ed331247.png)